framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -413,21 +414,83 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
                 factory.getDeclAnnotationWithMetaAnnotation(methodElement, metaAnnotation);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "value", String.class, false);
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
+
+            result.addAll(
+                    getPostConditionAnnotations(anno, metaAnno, annotationmirrorType, stringType));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns postcondition annotations.
+     *
+     * @param anno the type qualifier that {@code methodElement} should have
+     * @param metaAnno postCondition meta-annotation
+     * @param annotationmirrorType the annotationMirror class type used to cast the element value
+     * @param stringType the string class type used to cast the element value
+     */
+    private Set<Postcondition> getPostConditionAnnotations(
+            AnnotationMirror anno,
+            AnnotationMirror metaAnno,
+            Class<AnnotationMirror> annotationmirrorType,
+            Class<String> stringType) {
+
+        @SuppressWarnings("unchecked")
+        List<AnnotationValue> la =
+                AnnotationUtils.getElementValue(anno, "value", List.class, false);
+
+        Set<Postcondition> result = new LinkedHashSet<>();
+
+        // Check for multiple contracts.
+        List<AnnotationMirror> annotations = new ArrayList<>(la.size());
+        for (AnnotationValue a : la) {
+            if (a.getClass().getName().equals("javax.lang.model.element.AnnotationMirror")
+                    || a.getClass()
+                            .getName()
+                            .equals("com.sun.tools.javac.code.Attribute$Compound")) {
+                annotations.add(annotationmirrorType.cast(a.getValue()));
+            }
+        }
+
+        for (AnnotationMirror a : annotations) {
+            List<String> expression =
+                    AnnotationUtils.getElementValueArray(a, "value", String.class, false);
+            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, a);
             if (postcondAnno == null) {
                 continue;
             }
+            for (String expr : expression) {
+                result.add(new Postcondition(expr, postcondAnno, a));
+            }
+        }
+
+        // Check for a single contract.
+        List<String> expressions = new ArrayList<>(la.size());
+
+        for (AnnotationValue a : la) {
+            if (a.getClass().getName().equals("java.lang.String")
+                    || a.getClass()
+                            .getName()
+                            .equals("com.sun.tools.javac.code.Attribute$Constant")) {
+                expressions.add(stringType.cast(a.getValue()));
```

Comment:
How is this working? The cast will work if it is a `String`. But if it is a `Attribute$Constant` why would a cast to the final class `String` work?

Reply:
> How is this working? The cast will work if it is a `String`. But if it is a `Attribute$Constant` why would a cast to the final class `String` work?

@wmdietl The value for an annotation element of String is `Attribute$Constant`([link](https://static.javadoc.io/org.kohsuke.sorcerer/sorcerer-javac/0.11/com/sun/tools/javac/code/Attribute.Constant.html)).
So if  `a` is `Attribute$Constant` then we should cast to String.
P.S.- We do not need to check that if the value class is `String`. I have updated the code.
Please review it.